crates/devolutions-pedm/src/api/mod.rs

```diff
@@ -91,23 +91,23 @@ async fn named_pipe_middleware(
 fn create_pipe(pipe_name: &'static str) -> Result<NamedPipeServer> {
     let pipe = ServerOptions::new().write_dac(true).create(pipe_name)?;
 
-    let dacl = Acl::with_aces(vec![
-        Ace {
-            flags: ACE_FLAGS(0),
-            access_mask: GENERIC_READ.0 | GENERIC_WRITE.0,
-            data: AceType::AccessAllowed(Sid::from_well_known(WinBuiltinUsersSid, None)?),
+    let dacl = Acl::new()?.set_entries(&[
+        ExplicitAccess {
+            access_permissions: GENERIC_READ.0 | GENERIC_WRITE.0,
+            access_mode: GRANT_ACCESS,
+            inheritance: NO_INHERITANCE,
+            trustee: Trustee::Sid(Sid::from_well_known(WinBuiltinUsersSid, None)?),
         },
-        Ace {
-            flags: ACE_FLAGS(0),
-            access_mask: PIPE_ACCESS_FULL_CONTROL,
-            data: AceType::AccessAllowed(Token::current_process_token().sid_and_attributes()?.sid),
+        ExplicitAccess {
+            access_permissions: PIPE_ACCESS_FULL_CONTROL,
+            access_mode: GRANT_ACCESS,
+            inheritance: NO_INHERITANCE,
+            trustee: Trustee::Sid(Token::current_process_token().sid_and_attributes()?.sid),
         },
-    ])
-    .to_raw()?;
+    ])?;
```

Comment:
@thenextman I moved away from building and adding ACE structures to the ACL, and instead `SetEntriesInAclW` is used along `EXPLICIT_ACCESS_W` structures. It seems to be equivalent.

Reply:
Excellent!